### ISSUE TITLE
gawk segfault

### ISSUE THREAD

**Issue by joelteon** [NixOS/nixpkgs#6554]
On OSX Yosemite, gawk segfaults during build:

```
/bin/bash: line 1: 33468 Abort trap: 6           LC_ALL=${GAWKLOCALE:-C} LANG=${GAWKLOCALE:-C} AWKLIBPATH=../extension/.libs ../gawk -f ./readdir.awk .. > _readdir
make[2]: *** [readdir] Error 134
make[1]: *** [shlib-tests] Error 2
make: *** [check-recursive] Error 1
```

Problem details here: https://gist.github.com/joelteon/3b07197daee6a8be1699

Reverting dad86b4f544db4d497e4bbe769dec0bb3e9cfbd7 fixes the problem.

**Comment by joelteon** [comment]
This prevents installation of any package on all Darwin systems.

**Comment by jcf** [comment]
@joelteon is there a workaround I can use until this is fixed?

**Comment by jcf** [comment]
Trying out the following massive hack now:

``` sh
sudo chmod -x /usr/bin/xcrun
```

**Comment by jcf** [comment]
Reverting dad86b4 as suggested does appear to fix the problem. This was easier to do than I initially realised because I didn't realise the commit that needs to be reverted is in the nixpkgs repo I cloned to my machine during the setup process.

Thanks @joelteon!

**Comment by joelteon** [comment]
Fixed by ec019485d6b84d680d1a91a8f7ac8d94a57c95f4. I know dad86b4f544db4d497e4bbe769dec0bb3e9cfbd7 was committed for a reason but the tradeoff of breaking nixpkgs completely on Yosemite isn't worth it.

**Comment by gridaphobe** [comment]
3ea2aa4136f4f926e4a981011041564590f84450 fixes this issue in a more portable manner.
